server/src/main/java/org/opensearch/rest/action/admin/cluster/RestNodesStatsAction.java

```diff
@@ -189,6 +189,13 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         if (nodesStatsRequest.indices().isSet(Flag.Segments)) {
             nodesStatsRequest.indices().includeSegmentFileSizes(request.paramAsBoolean("include_segment_file_sizes", false));
         }
+        if (request.hasParam("include_all")) {
+            nodesStatsRequest.indices().includeAllShardIndexingPressureTrackers(request.paramAsBoolean("include_all", false));
+        }
+
+        if (request.hasParam("top")) {
+            nodesStatsRequest.indices().includeOnlyTopIndexingPressureMetrics(request.paramAsBoolean("top", false));
+        }
```

Comment:
```suggestion
        if (request.hasParam("include_all")) {
            nodesStatsRequest.indices().includeAllShardIndexingPressureTrackers(request.paramAsBoolean("include_all", false));
        }

        if (nodesStatsRequest.indices().isSet(Flag.Indexing) && request.hasParam("top")) {
            nodesStatsRequest.indices().includeOnlyTopIndexingPressureMetrics(request.paramAsBoolean("top", false));
        }
```

Reply:
This is not tied up with ``index_metric`` or ``indices`` metrics, so not sure why is this needed.

Reply:
what does top and include all param mean? would it only be applicable for shard indexing pressure

Reply:
``top`` is to return the top level aggregated shard indexing pressure stats on the node, while ``include_all `` returns detailed stats from all the trackers across hot and cold stores. Yes these are only applicable for shard indexing pressure currently.

Reply:
Just for my understanding how is the endpoint supposed to look like with top?

Reply:
It will look like :

```
GET /_nodes/_local/stats/shard_indexing_pressure?top
GET /_nodes/stats/shard_indexing_pressure?top
```

Reply:
Then can we check if this `top` is being used only the context of `shard_indexing_pressure `

Reply:
Handling of stats query params is separate and at a common place than from the requested node-stats metric. For others it will be a no-op similar to other params.